docs/reference/search/retriever.asciidoc

```diff
@@ -0,0 +1,225 @@
+[[retriever]]
+=== Retriever API
+
+preview::["This functionality is in technical preview and may be changed or removed in a future release. The syntax will likely change before GA. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features."]
+
+A retriever is a specification to describe top documents returned from a
+search. A retriever replaces other elements of the <<search-search, search API>>
+that also return top documents such as <<query-dsl, `query`>> and
+<<search-api-knn, `knn`>>. A retriever may have child retrievers where a
+retriever with two or more children is considered a compound retriever. This
+allows for complex behavior to be depicted in a tree-like structure, called
+the retriever tree, to better clarify the order of operations that occur
+during a search.
+
+The following retrievers are available:
+
+`standard`::
+A <<standard-retriever, retriever>> that replaces the functionality of a traditional <<query-dsl, query>>.
+
+`knn`::
+A <<knn-retriever, retriever>> that replaces the functionality of a <<search-api-knn, knn search>>.
```

Comment:
I could see this being a little confusing to users, since knn is also a query (e.g. could knn-query be nested under a standard retriever?) I realize this is a weird case, but it might be good to address this somewhere, maybe in a blog post or something? WDYT?

Reply:
You could use a knn query under a standard retriever. I agree this is probably better left to documentation in a different place, though. I worry that adding some notes about how you can use knn-query as part of a standard retriever will just confuse the issue more currently. I'll give this some thought as a possible follow up.